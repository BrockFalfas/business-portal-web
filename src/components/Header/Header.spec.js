import React from 'react';
import { shallow } from 'enzyme';

import { Header } from './index';

const initHeader = overrides => shallow(<Header {...overrides} />);

describe('component: <Header/>', () => {
  it('should render without crashing', () => {
    const wrapper = initHeader();
    expect(wrapper).toBeTruthy();
  });

  it('should render proper header for different sizes', () => {
    const allowedSizes = Object.entries(Header.headerSizes);
    allowedSizes.forEach(([size, componentName]) => {
      const wrapper = initHeader({ size });
      expect(wrapper.find(`${componentName}.Header-title`).length).toBe(1);
    });
  });

  it('should render children next to the header', () => {
    const fakeChild = <div>fake child</div>;
    const wrapper = initHeader({ children: fakeChild });
    expect(wrapper).toMatchSnapshot();
  });
});
