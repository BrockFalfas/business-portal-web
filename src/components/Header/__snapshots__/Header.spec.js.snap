// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <Header/> should render children next to the header 1`] = `
<header
  className="Header"
>
  <h2
    className="Header-title"
  />
  <div>
    fake child
  </div>
</header>
`;
